interface/ceed-basis.c

```diff
@@ -1530,7 +1576,8 @@ int CeedBasisApplyAtPoints(CeedBasis basis, CeedInt num_points, CeedTransposeMod
   }
 
   // Default implementation
-  CeedCheck(basis->is_tensor_basis, basis->ceed, CEED_ERROR_UNSUPPORTED, "Evaluation at arbitrary points only supported for tensor product bases");
+  CeedCall(CeedBasisIsTensor(basis, &is_tensor_basis));
+  CeedCheck(is_tensor_basis, ceed, CEED_ERROR_UNSUPPORTED, "Evaluation at arbitrary points only supported for tensor product bases");
```

Comment:
Since the "boolean return" functions aren't checking their pointers, couldn't we have then direction return the boolean? So we could instead have:

```suggestion
  CeedCheck(CeedBasisIsTensor(basis), ceed, CEED_ERROR_UNSUPPORTED, "Evaluation at arbitrary points only supported for tensor product bases");
```

Reply:
I guess that would prevent any future checking of the pointers and returning a error. But we could also simply check pointers at the top of the function too.

Reply:
I'm not opposed to the idea, but that would be a breaking change, so it should be discussed in an issue and put in a separate PR